test/core/reporters/helpers/process-aggregate.js

```diff
@@ -96,12 +135,17 @@ describe('helpers.processAggregate', function () {
 
     describe('`resultTypes` option', function () {
 
-      it('should remove non-specified result types from the `resultObject`', function () {
-        var resultObject = helpers.processAggregate(results, { resultTypes: ['passes', 'violations'] });
+      it('should reduce the unwanted result types to 1 in the `resultObject`', function () {
+        var resultObject = helpers.processAggregate(results, { resultTypes: ['violations'] });
         assert.isDefined(resultObject.passes);
+        assert.equal(resultObject.passes[0].nodes.length, 1);
         assert.isDefined(resultObject.violations);
-        assert.isUndefined(resultObject.incomplete);
-        assert.isUndefined(resultObject.inapplicable);
+        assert.equal(resultObject.violations[0].nodes.length, 2);
+        resultObject = helpers.processAggregate(results, { resultTypes: ['passes'] });
```

Comment:
Why the second call to `processAggregate` here? Is that what happens when it gets called as a result of an analysis?

Reply:
It's just a second test to make sure it's not just 'violations', but also that it works for 'passes'.